broker-core/src/test/java/io/zeebe/broker/incident/ExpressionIncidentTest.java

```diff
@@ -293,7 +293,7 @@ public void shouldResolveIncidentIfInstanceCanceled() {
     assertThat(incidentEvent.getKey()).isGreaterThan(0);
     assertIncidentRecordValue(
         ErrorType.CONDITION_ERROR.name(),
-        "Cannot compare values of different types: STRING and INTEGER",
+        "Failed to evaluate condition '$.foo >= 5 && $.foo < 10', reason: Cannot compare values of different types: STRING and INTEGER",
```

Comment:
```suggestion
        "Expected to evaluate condition '$.foo >= 5 && $.foo < 10' successfully, but failed because: Cannot compare values of different types: STRING and INTEGER",
```